wondrous-app/components/Members/hooks.tsx

```diff
@@ -0,0 +1,245 @@
+import { useEffect, useState } from 'react';
+import { useLazyQuery } from '@apollo/client';
+import * as Sentry from '@sentry/nextjs';
+import isEqual from 'lodash/isEqual';
+import { GET_ORG_MEMBERSHIP_REQUEST, GET_ORG_USERS, GET_POD_MEMBERSHIP_REQUEST, GET_POD_USERS } from 'graphql/queries';
+import { QUERY_LIMIT } from './constants';
+
+export const useGetOrgMemberRequests = (orgId, searchString = '', roleIds = []) => {
+  const [hasMore, setHasMore] = useState(false);
+  const [isInitialFetchForThePage, setIsInitialFetchForThePage] = useState(true); // this state is used to determine if the fetch is from a fetchMore or from route change
+  const [previousSearchQuery, setPreviousSearchQuery] = useState('');
+  const [previousRoleFilter, setPreviousRoleFilter] = useState([]);
+
+  const [getOrgUserMembershipRequests, { data, fetchMore, previousData, variables }] = useLazyQuery(
+    GET_ORG_MEMBERSHIP_REQUEST,
+    {
+      fetchPolicy: 'network-only',
+      // set notifyOnNetworkStatusChange to true if you want to trigger a rerender whenever the request status updates
+      notifyOnNetworkStatusChange: true,
+      onCompleted: ({ getOrgMembershipRequest }) => {
```

Comment:
i think this logic is way too complicated, we can probably make this better @Lamperoyge

Reply:
fixed this in the latest code @terryli0095